tests/unit/experimental/pipeline/components.py

```diff
@@ -41,3 +42,16 @@ async def run(self, number1: int, number2: int) -> IntResultModel:
 class ComponentMultiply(Component):
     async def run(self, number1: int, number2: int = 2) -> IntResultModel:
         return IntResultModel(result=number1 * number2)
+
+
+class ComponentMultiplyWithContext(Component):
+    async def run(self, number1: int, number2: int) -> IntResultModel:
+        return IntResultModel(result=number1 * number2)
+
+    async def run_with_context(
```

Comment:
So one always has to implement both `run` and `run_with_context`?

Reply:
Hum, yeah, that's definitely not nice. I've updated the `Component` implementation to remove this. `Component` is not an abstract class anymore, but we check that either `run` or `run_with_context` is implemented, and take the later as reference to parse the expected inputs/outputs.